clippy_lints/src/entry.rs

```diff
@@ -166,7 +166,11 @@ impl<'tcx> LateLintPass<'tcx> for HashMapPass {
                     "if let {}::{entry_kind} = {map_str}.entry({key_str}) {body_str}",
                     map_ty.entry_path(),
                 ))
-            } else if let Some(insertion) = then_search.as_single_insertion() {
+            } else if let Some(insertion) = then_search.as_single_insertion()
+                && let span_in_between = then_expr.span.shrink_to_lo().between(insertion.call.span)
+                && let span_in_between = span_in_between.with_lo(span_in_between.lo() + BytePos::from_usize(1))
```

Comment:
- `BytePos::from_usize(1)` could be replaced with `BytePos(1)`. In fact, this whole line could be simplified using `Span::split_at`:
```suggestion
                // remove the opening brace
                && let span_in_between = span_in_between.split_at(1).1
```
- I think the same check would be needed for the code _after_ the `insert`?

Generally, it could be better to do all these checks somewhere inside `InsertSearcher`? Though I'm not very familiar with visitors, so I'm not sure whether that's actually feasible.

Reply:
> I think the same check would be needed for the code _after_ the `insert`?

Uh, I think it's a bit more complicated than that -- if there is `cfg`d-out code after the `insert`, then this case isn't even `allow_insert_closure` anymore. I think this is another argument for doing this check already in the visitor